jenkins/sparklingWaterPipeline.groovy

```diff
@@ -220,4 +251,36 @@ def pysparklingIntegTest() {
     }
 }
 
+def publishNightly(){
+    return { config ->
+        stage ('Nightly: Publishing Artifacts to S3'){
+            if (config.buildNightly.toBoolean()) {
+
+                sh """
+                    # echo 'Making distribution'
+                    ${getGradleCommand(config)} buildSparklingWaterDist
+
+                    # Upload to S3
+                    """
+                def fullSWVersion = sh(
+                        script: "echo \$( cat gradle.properties | grep version | grep -v '#' | sed -e \"s/.*=//\" )",
+                        returnStdout: true).trim()
+
+                def majorSWVersion = fullSWVersion.toString().substring(0, fullSWVersion.toString().lastIndexOf('_'))
+                def buildSWVersion = fullSWVersion.toString().substring(fullSWVersion.toString().lastIndexOf('_') + 1)
+                def artifactPath = "${env.WORKSPACE}/dist/build/sparkling-water-${fullSWVersion}.zip"
+
+                s3up {
```

Comment:
This is good, but main place to upload Sparkling Water artifacts should
be the same place where releases are uploaded including web page.

Reply:
s3up is for new projects mainly....

Reply:
Ah, I see - the goals is to have nightly released in the same way as regular release except that we don't the the maven publish. I'll work on it.

I'm however a bit confused on `s3up` - was pointed to this by @michal-raska . Any alternative I could use ? :) ?

Reply:
`s3up` we wrote for dtbl, DAI. It does not fit into h2o-3 structure. for now i would use s3cmd directly.

Reply:
the plan is to slowly unify infrastructure tho share these pieces of code.

Reply:
Oki, thanks! I didn't want to duplicate code so that's why I went for s3up.

I will reuse the code for sparkling water release script as we want to have the same page and release for nightly.